src/gf.c

```diff
@@ -2500,9 +2511,8 @@ static int ml_matches_visitor(jl_typemap_entry_t *ml, struct typemap_intersectio
                         // the current method doesn't match if there is an intersection with an
                         // ambiguous method that covers our intersection with this one.
                         jl_value_t *ambi = jl_type_intersection_matching((jl_value_t*)ml->sig,
-                                                                         (jl_value_t*)mambig->sig,
-                                                                         &env, mambig->tvars);
-                        if (jl_subtype(closure->match.ti, ambi, 0)) {
+                                                                         (jl_value_t*)mambig->sig, &env);
+                        if (jl_subtype(closure->match.ti, ambi)) {
```

Comment:
We shouldn't need to examine `ml->sig` at all, since we already know that `typeintersect(ml->sig, types) <: ml->sig` (while `closure->match.ti` is an over-approximation, the runtime type-intersection will be precise).